### ISSUE TITLE
Do not support custom layout on restart

### ISSUE THREAD

**Issue by Adirelle** [gerbera/gerbera#232]
I have setup Gerbera as per the attached file. It uses a very simple, custom layout script, autoscan and no UI.

When I start it with no database, everything is fine. The directories are scanned and the items are properly found in the place I wanted (which mimics the filesystem directory), for example:

```
Vidéos
    Series
        SerieA
            EP01
            EP02
            EP03
            EP04
```

But when I restart the server with an existing database, it seems to re-organize some of the existing items (not all) as if no custom layout is configured, resulting in a messed up file hierarchy. Files from the same directory ends up either in the path configured through the scripts or in the path as per basic layout, for example:

```
SerieA
    EP02
    EP03
Vidéos
    Series
        SerieA
            EP01
            EP04
```

[config.xml.txt](https://github.com/gerbera/gerbera/files/1609111/config.xml.txt)
[virtual-layout.js.txt](https://github.com/gerbera/gerbera/files/1609112/virtual-layout.js.txt)

**Comment by Adirelle** [comment]
Actually, it seems the same thing happens using the "builtin" layout, so maybe it is more related to the autoscan entries than to the scripts.

**Comment by elmodaddyb** [comment]
During startup the server, a few things happen.
1. Content Manager created -- this retrieves the list of saved autoscans from the database
2. Content Manager initialized -- this notify's all timed autoscans

I see that autoscan is run, and results `ContentManager::initLayout()` which does include `JSLayout` support using the **import-script** specified in configuration xpath `/import/scripting/virtual-layout/import-script`.

I wonder if there is some config retrieval concurrency which returns the default type of **builtin**, instead of **js**.
```c++
temp = getOption(_("/import/scripting/virtual-layout/attribute::type"),
        _(DEFAULT_LAYOUT_TYPE));
```

**Comment by Adirelle** [comment]
As said earlier, messed up directories *also* happens when using `<virtual-layout type="builtin">`. So I am not sure about that very specific point, unless it also (somehow) uses the provided script in that case.

Another thing I was thinking about is: isn't Gerbera reloading data from the database and passing it again through the script ? If the script is not idempotent, it could lead to strange results.

**Comment by elmodaddyb** [comment]
Yea, sorry I misunderstood your first point.  I will try to reproduce the issue and debug.

**Comment by chiraag-nataraj** [comment]
I actually just ran into this issue, and it is _very much_ still an issue. Coupled with #448, using `gerbera` for serving video files is quite unwieldy.

**Comment by v00d00** [comment]
@chiraag-nataraj Can you give some suggestion on how to reproduce the issue?

**Comment by chiraag-nataraj** [comment]
Sorry for the delay.
All I did was the following:
Setup: I have a TV show ("The Son") which I bought on DVD. It came on 3 DVDs, so I organized it into the following hierarchy:
```
The Son/
The Son/Disc 1/
The Son/Disc 2/
The Son/Disc 3/
```
with the appropriate episodes in each directory (1-4 in `Disc 1`, 5-8 in `Disc 2`, and 9 and 10 in `Disc 3`).
1. Copy everything over in the directory structure indicated above (in my case, I copied it into `/media/files/Videos/`, and `/media/files` is the root of my media library).
2. Add `/media/files/Videos` to the Autoscan directory list through the web interface and set up a full scan.
3. Go back to the database view and note that everything looks great - the hierarchy mentioned above is replicated perfectly in the database view and everything is clearly organized.
4. Now restart `gerbera`.
5. Note that after a little bit, various other folders (titled `Disc 1`, `Disc 2`, and so on) start appearing.
6. Now, some of the episodes 1-4 are in the `Disc 1` folder and others are in `The Son/Disc 1` in the Database view (nothing's changed on disk).

I got frustrated enough with this that I ended up getting rid of the hierarchy entirely and just put everything inside `The Son` and now everything is fine.
